frame/nfts/src/features/atomic_swap.rs

```diff
@@ -80,30 +99,34 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		};
 
 		if !is_past_deadline {
-			ensure!(item.owner == sender, Error::<T, I>::NoPermission);
+			let item = Item::<T, I>::get(&offered_collection_id, &offered_item_id)
+				.ok_or(Error::<T, I>::UnknownItem)?;
+			ensure!(item.owner == caller, Error::<T, I>::NoPermission);
 		}
 
-		PendingSwapOf::<T, I>::remove(&collection_id, &item_id);
+		PendingSwapOf::<T, I>::remove(&offered_collection_id, &offered_item_id);
 
 		Self::deposit_event(Event::SwapCancelled {
-			collection: collection_id,
-			item: item_id,
+			offered_collection: offered_collection_id,
+			offered_item: offered_item_id,
 			desired_collection: swap.desired_collection,
 			desired_item: swap.desired_item,
 			price: swap.price,
+			price_direction: swap.price_direction,
 			deadline: swap.deadline,
 		});
 
 		Ok(())
 	}
 
 	pub fn do_claim_swap(
-		sender: T::AccountId,
+		caller: T::AccountId,
 		send_collection_id: T::CollectionId,
 		send_item_id: T::ItemId,
 		receive_collection_id: T::CollectionId,
 		receive_item_id: T::ItemId,
-		maybe_receive_amount: Option<ItemPrice<T, I>>,
+		maybe_price: Option<ItemPrice<T, I>>,
+		maybe_price_direction: Option<PriceDirection>,
```

Comment:
just wondering if the fields should get rolled up into a `SwapWitness` struct that holds the data about the swap that the caller is using to identify that particular swap?

Reply:
hmm...semantically it would make sense to add such a field.
I'm thinking here how could it look like, smth like `SwapWitness{ price: PriceWithDirection {amount, direction} }`?

Reply:
That works.

Reply:
renamed to `witness_price`